python/decord/audio_reader.py

```diff
@@ -0,0 +1,168 @@
+"""Audio Reader."""
+from __future__ import absolute_import
+
+import ctypes
+import numpy as np
+import math
+
+from ._ffi.base import c_array, c_str
+from ._ffi.function import _init_api
+from ._ffi.ndarray import DECORDContext
+from .base import DECORDError
+from . import ndarray as _nd
+from .ndarray import cpu, gpu
+from .bridge import bridge_out
+
+AudioReaderHandle = ctypes.c_void_p
+
+class AudioReader(object):
+    """Individual audio reader with convenient indexing function.
+
+    Parameters
+    ----------
+    uri: str
+        Path of file.
+    ctx: decord.Context
+        The context to decode the file, can be decord.cpu() or decord.gpu().
+    sample_rate: int, default is -1
+        Desired output sample rate of the audio, unchanged if `-1` is specified.
+    mono: bool, default is True
+        Desired output channel layout of the audio. `True` is mono layout. `False` is unchanged.
+
+    """
+
+    def __init__(self, uri, ctx=cpu(0), sample_rate=-1, mono=True):
+        self._handle = None
+        assert isinstance(ctx, DECORDContext)
+        is_mono = 1 if mono else 0
+        if hasattr(uri, 'read'):
+            ba = bytearray(uri.read())
+            uri = '{} bytes'.format(len(ba))
+            self._handle = _CAPI_AudioReaderGetAudioReader(
+                ba, ctx.device_type, ctx.device_id, sample_rate, 2, is_mono)
+        else:
+            self._handle = _CAPI_AudioReaderGetAudioReader(
+                uri, ctx.device_type, ctx.device_id, sample_rate, 0, is_mono)
+        if self._handle is None:
+            raise RuntimeError("Error reading " + uri + "...")
+        self._array = _CAPI_AudioReaderGetNDArray(self._handle)
+        # self._array = bridge_out(self._array)
+        self._array = self._array.asnumpy()
+        self._duration = _CAPI_AudioReaderGetDuration(self._handle)
+        self._num_samples_per_channel = _CAPI_AudioReaderGetNumSamplesPerChannel(self._handle)
+        self._num_channels = _CAPI_AudioReaderGetNumChannels(self._handle)
+        self.sample_rate = sample_rate
+
+    def __del__(self):
+        if self._handle:
+            _CAPI_AudioReaderFree(self._handle)
+
+    def __getitem__(self, idx):
+        """Get sample at `idx`.
```

Comment:
`idx` is the index of resampled audio, unit is `sample`

Reply:
Do you mean I should change idx to be sample?